GliaWidgets/Sources/CallVisualizer/ScreenSharing/View/ScreenSharingView.swift

```diff
@@ -1,127 +1,59 @@
-            }
-
-            setFontScalingEnabled(
-                props.style.accessibility.isFontScalingEnabled,
-                for: messageLabel
-            )
-            setFontScalingEnabled(
-                props.style.buttonStyle.accessibility.isFontScalingEnabled,
-                for: endScreenSharingButton
-            )
-        }
-
-        override func layoutSubviews() {
-            switch props.style.backgroundColor {
-            case .fill(let color):
-                backgroundColor = color
-            case .gradient(let colors):
-                makeGradientBackground(colors: colors)
+    struct ScreenSharingView: View {
+        @ObservedObject var model: Model
+
+        var body: some View {
+            ZStack {
+                Background(model.style.backgroundColor)
+                    .edgesIgnoringSafeArea(.all)
+                VStack(spacing: 0) {
+                    HeaderSwiftUI(model: model.makeHeaderModel())
+                    VStack(spacing: 16) {
+                        mainLabel
+                        endScreenShareButton
+                    }
+                    .frame(maxWidth: .infinity, maxHeight: .infinity)
+                }.edgesIgnoringSafeArea(.top)
             }
         }
     }
 }
 
-// MARK: - Private
+extension CallVisualizer.ScreenSharingView {
+    var mainLabel: some View {
+        Text(model.style.messageText)
+            .font(.convert(model.style.messageTextFont))
+            .foregroundColor(SwiftUI.Color(model.style.messageTextColor))
+            .lineLimit(2)
+            .multilineTextAlignment(.center)
+            .accessibility(identifier: "end_screen_sharing_message")
+    }
 
-private extension CallVisualizer.ScreenSharingView {
-    func renderProps() {
-        header.props = props.header
-        endScreenSharingButton.props = props.endScreenSharing
+    var endScreenShareButton: some View {
+        SwiftUI.Button(action: {
+            model.event(.endScreenShareTapped)
+        }, label: {
+            HStack(spacing: 8) {
+                SwiftUI.Image(uiImage: model.style.buttonIcon)
+                    .resizable()
+                    .aspectRatio(contentMode: .fit)
+                    .frame(height: 24)
+                    .foregroundColor(SwiftUI.Color(model.style.buttonStyle.titleColor))
+                Text(model.style.buttonStyle.title)
+                    .font(.convert(model.style.buttonStyle.titleFont))
+                    .foregroundColor(SwiftUI.Color(model.style.buttonStyle.titleColor))
+                    .lineLimit(nil)
+            }
+            .padding(.horizontal, 4)
+            .frame(maxWidth: .infinity, minHeight: 40, idealHeight: 40)
+            .background(Background(model.style.buttonStyle.backgroundColor))
+            .cornerRadius(4)
+            .padding(.horizontal, 60)
+        })
+        .accessibility(identifier: "end_screen_sharing_button")
```

Comment:
Do these accessibility related modifiers produce deprecation warnings? Maybe it makes sense to use `migrationAccessibilityIdentifier` + others (for label and hint) until we properly backport missing features in SwiftUI for iOS 13?

Reply:
fixed for specific example and for all the other accessibility [bb2ef12](https://github.com/salemove/ios-sdk-widgets/pull/784/commits/bb2ef12430273c7e0f656c63dc5932866e839ad0)

Reply:
And here [860c457](https://github.com/salemove/ios-sdk-widgets/pull/784/commits/860c4579ff9324273ca81628bd583a841935996d)